src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -0,0 +1,95 @@
+
+        public ServerHandler(ConnectionService connectionService) : base(connectionService)
+        {
+        }
+
+        public override bool CanHandleType(SqlObjectType objectType)
+        {
+            return objectType == SqlObjectType.Server;
+        }
+
+        public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRequestParams requestParams)
+        {
+            ConnectionInfo connInfo = this.GetConnectionInfo(requestParams.ConnectionUri);
+            ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementService.ApplicationName);
+
+            using (var context = new ServerViewContext(requestParams, serverConnection))
+            {
+                this.server = new Server(context.Connection); ;
+                if(this.server != null)
+                {
+                    this.serverViewInfo.ObjectInfo = new ServerInfo()
+                    {
+                        Name = server.Name,
+                        HardwareGeneration = server.HardwareGeneration,
+                        Language = server.Language,
+                        MemoryInMB = server.PhysicalMemory,
+                        OperatingSystem = server.HostDistribution,
+                        Platform = server.HostPlatform,
+                        Processors = server.Processors,
+                        IsClustered = server.IsClustered,
+                        IsHadrEnabled = server.IsHadrEnabled,
+                        IsPolyBaseInstalled = server.IsPolyBaseInstalled,
+                        IsXTPSupported = server.IsXTPSupported,
+                        Product = server.Product,
+                        ReservedStorageSizeMB = server.ReservedStorageSizeMB,
+                        RootDirectory = server.RootDirectory,
+                        ServerCollation = server.Collation,
+                        ServiceTier = server.ServiceTier,
+                        StorageSpaceUsageInGB = (int)ByteConverter.ConvertMbtoGb(server.UsedStorageSizeMB),
+                        Version = server.Version.ToString(),
+                        MinServerMemory = GetServerMinMemory(),
+                        MaxServerMemory = GetServerMaxMemory()
+                    };
+                }
+
+                return Task.FromResult(new InitializeViewResult { ViewInfo = this.serverViewInfo, Context = context });
+            }
+        }
+
+        public override Task Save(ServerViewContext context, ServerInfo serverInfo)
+        {
+            throw new NotSupportedException("ServerHandler does not support Save method");
+        }
+
+        public override Task<string> Script(ServerViewContext context, ServerInfo obj)
+        {
+            throw new NotSupportedException("ServerHandler does not support Script method");
+        }
+
+        private int GetServerMaxMemory()
+        {
+            return configService.GetSmoConfig(server, 1544).ConfigValue;
```

Comment:
Please make constants for these numbers (I'd add them in the config service file)